types/xdg_shell/wlr_xdg_shell.c

```diff
@@ -170,3 +186,37 @@ void wlr_xdg_shell_destroy(struct wlr_xdg_shell *xdg_shell) {
 	wl_global_destroy(xdg_shell->wl_global);
 	free(xdg_shell);
 }
+
+static void wlr_surface_get_bounding_box(const struct wlr_surface *surface, struct wlr_box *box) {
+	int min_x = 0, min_y = 0;
+	int max_x = surface->current->width;
+	int max_y = surface->current->height;
+
+	struct wlr_subsurface *sub;
+	wl_list_for_each(sub, &surface->subsurfaces, parent_link) {
```

Comment:
This isn't going to work for nested subsurfaces. Use surface iterators instead (`wlr_xdg_surface_for_each_surface`).

Reply:
Urgh I forgot about those.
But it should be `wlr_surface_for_each_surface` so we don't consider popups

Reply:
Right